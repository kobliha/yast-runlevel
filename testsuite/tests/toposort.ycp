// test for TopologicalSort
{
    include "testsuite.ycp";
    include "runlevel/toposort.ycp";

    DUMP ("TopologicalSort");

    map g_empty = $[];
    DUMP (TopologicalSort (g_empty));

    map g_loop = $[
	"v1": [ "v1", ],
	];
    DUMP (TopologicalSort (g_loop));

    map g = $[
	"v1": [ "v2", "v3", ],
	"v2": [ "v3", "v4", ],
	"v3": [ "v6", ],
	"v4": [ "v6", ],
	"v5": [ "v4", ],
	"v6": [ ],
	"v7": [ "v6", ],
	"v8": [ "v9", ],
	"v9": [ ],
	"v10": [ ],
	];
    DUMP (TopologicalSort (g));

    DUMP ("ReachableSubgraph");
    DUMP (ReachableSubgraph (g, "v1"));

    DUMP ("ReverseGraph");
    DUMP (ReverseGraph (g));
}
