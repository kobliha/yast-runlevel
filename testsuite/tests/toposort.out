Dump	TopologicalSort
Dump	[[], []]
Log	Cyclic subgraph found, remainder has 1 nodes: ["v1"]
Dump	[[], ["v1"]]
Dump	[["v1", "v10", "v5", "v7", "v8", "v2", "v9", "v3", "v4", "v6"], []]
Dump	ReachableSubgraph
Dump	$["v1":["v2", "v3"], "v2":["v3", "v4"], "v3":["v6"], "v4":["v6"], "v6":[]]
Dump	ReverseGraph
Dump	$["v1":[], "v10":[], "v2":["v1"], "v3":["v1", "v2"], "v4":["v2", "v5"], "v5":[], "v6":["v3", "v4", "v7"], "v7":[], "v8":[], "v9":["v8"]]
